legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java

```diff
@@ -542,13 +542,9 @@ public static Object _evaluateToMany(ExecutionSupport es, Bridge bridge, org.fin
                 return CompiledSupport.executeMethod(o.getClass(), func._functionName(), func, Arrays.copyOfRange(paramClasses, 1, paramClasses.length),
                     o, Arrays.copyOfRange(paramInstances, 1, paramInstances.length), es);
             }
-            if (func instanceof ConcreteFunctionDefinition)
+            if (func instanceof NativeFunction || func instanceof LambdaFunction || func instanceof ConcreteFunctionDefinition)
             {
-                return CompiledSupport.executeFunction(func, paramClasses, paramInstances, es);
-            }
-            if (func instanceof NativeFunction || func instanceof LambdaFunction)
-            {
-                SharedPureFunction<?> foundFunc = getNativeOrLambdaFunction(es, func);
+                SharedPureFunction<?> foundFunc = findSharedPureFunction(func, bridge, es);
```

Comment:
Does findSharedPureFunction handle NativeFunction properly?

Reply:
Yes

Lamba functions are handled at https://github.com/finos/legend-pure/blob/3965e4e69af087750b5d9f6dcec8b8ef35863538/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java#L385

Native functions are handled at https://github.com/finos/legend-pure/blob/3965e4e69af087750b5d9f6dcec8b8ef35863538/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java#L434 which is the same logic as at https://github.com/finos/legend-pure/blob/3965e4e69af087750b5d9f6dcec8b8ef35863538/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java#L583